assets/js/modules/analytics-4/components/widgets/TopCitiesWidget.stories.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * TopCitiesWidget component stories.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { MODULES_ANALYTICS_4 } from '../../datastore/constants';
+import {
+	provideKeyMetrics,
+	provideModules,
+} from '../../../../../../tests/js/utils';
+import { withWidgetComponentProps } from '../../../../googlesitekit/widgets/util';
+import { getAnalytics4MockResponse } from '../../utils/data-mock';
+import { replaceValuesInAnalytics4ReportWithZeroData } from '../../../../../../.storybook/utils/zeroReports';
+import WithRegistrySetup from '../../../../../../tests/js/WithRegistrySetup';
+import TopCitiesWidget from './TopCitiesWidget';
+
+const reportOptions = {
+	startDate: '2020-08-11',
+	endDate: '2020-09-07',
+	dimensions: [ 'city' ],
+	metrics: [ { name: 'totalUsers' } ],
+	orderby: [
+		{
+			metric: {
+				metricName: 'totalUsers',
+			},
+			desc: true,
+		},
+	],
+	limit: 3,
+};
+
+const WidgetWithComponentProps =
+	withWidgetComponentProps( 'test' )( TopCitiesWidget );
+
+const Template = ( { setupRegistry, ...args } ) => (
+	<WithRegistrySetup func={ setupRegistry }>
+		<WidgetWithComponentProps { ...args } />
+	</WithRegistrySetup>
+);
+
+export const Ready = Template.bind( {} );
+Ready.storyName = 'Ready';
+Ready.args = {
+	setupRegistry: ( registry ) => {
+		const report = getAnalytics4MockResponse( reportOptions );
+		// Calculate sum of metricValues for all rows
+		const rowsSum = report.rows.reduce( ( total, row ) => {
+			return total + Number( row.metricValues[ 0 ].value );
+		}, 0 );
+
+		// Generate totalValueForAllCities that is higher than the sum
+		const totalValueForAllCities = rowsSum * 2;
+
+		// Adjust totals field in the mock response
+		report.totals = [
+			{
+				dimensionValues: [ { value: 'RESERVED_TOTAL' } ],
+				metricValues: [ { value: totalValueForAllCities.toString() } ],
+			},
+		];
+		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetReport( report, {
+			options: reportOptions,
+		} );
+	},
+};
+Ready.scenario = {
+	label: 'KeyMetrics/TopCitiesWidget/Ready',
+	delay: 250,
```

Comment:
Let's not add `delay` unless it's really needed for some reason, in which case let's annotate why.